drivers/wireless/lpwan/rn2483/rn2483.c

```diff
@@ -0,0 +1,862 @@
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set crc (Cyclic redundancy check, true for enabled)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_CRC, &priv->config.crc);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set pwr (Transmit output power)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_PWR, &priv->config.pwr);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set sf (Spread Factor)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_SF, &priv->config.sf);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set sync (Synchronization word)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_SYNC, &priv->config.sync);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set mod (Modulation: FSK or LoRa)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_MOD, &priv->config.mod);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  return F_OK;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: rn2483_register
+ *
+ * Description:
+ *   Register the RN2483 LoRa driver.
+ *
+ ****************************************************************************/
+
+int rn2483_register(FAR const char *devpath, FAR const char *uartpath)
+{
+  syslog(LOG_INFO, "Enter Register\n\n");
+
+  FAR struct rn2483_dev_s *priv = NULL;
+  int err = 0;
+
+  DEBUGASSERT(uartpath != NULL);
+
+  /* Initialize the device structure */
+
+  priv = kmm_zalloc(sizeof(struct rn2483_dev_s));
+  if (priv == NULL)
+  {
+    snerr("ERROR: Failed to allocate instance.\n");
+    return -ENOMEM;
+  }
+
+  /* Initialize mutex */
+
+  err = nxmutex_init(&priv->devlock);
+  if (err < 0)
+  {
+    // TODO: what logging macro needs to be used
+    // snerr("ERROR: Failed to register SHT4X driver: %d\n", err);
+    kmm_free(priv);
+    return err;
+  }
+
+  /* Get access to underlying UART interface */
+
+  err = file_open(&priv->uart, uartpath, O_RDWR | O_CLOEXEC);
+  if (err < 0)
+  {
+    // TODO log error
+    nxmutex_destroy(&priv->devlock);
+    kmm_free(priv);
+    return err;
+  }
+
+  /* Set configuration options. */
+
+#if CONFIG_LPWAN_RN2483_CR == 5
+  priv->config.cr = RN2483_CR_4_5;
+#elif CONFIG_LPWAN_RN2483_CR == 6
+  priv->config.cr = RN2483_CR_4_6;
+#elif CONFIG_LPWAN_RN2483_CR == 7
+  priv->config.cr = RN2483_CR_4_7;
+#elif CONFIG_LPWAN_RN2483_CR == 8
+  priv->config.cr = RN2483_CR_4_8;
+#endif
+
+#if CONFIG_LPWAN_RN2483_LORA
+  priv->config.mod = RN2483_MOD_LORA;
+#else
+  priv->config.mod = RN2483_MOD_FSK;
+#endif
+
+  priv->config.freq = CONFIG_LPWAN_RN2483_FREQ;
+
+  // Error check of frequency value (Not sure if this is what you want Matteo)
+  assert((priv->config.freq >= 433000000 && priv->config.freq <= 434800000) || (priv->config.freq >= 863000000 && priv->config.freq <= 870000000));
```

Comment:
You should perform this check using the `#define` macros at the top of the file, similar to how it was done for other parameters.

```c
#if CONFIG_LPWAN_RN2493_FREQ < 433000000 || CONFIG_LPWAN_RN2483_FREQ > 4348000000
#error "Frequency must be between ....<err msg here>"
#endif
```
You've done that which is correct! Now there is no way that in the register function the parameters are faulty/invalid, so the assert is not needed.

Runtime checking only needs to be done in the `ioctl` function when parameters are set after compilation and while the driver is running. Those you need to do in code, similar to what you've done within the assert statement, but instead of crashing due to a faulty assertion you should return `-EINVAL`.